src/shared/components/XYPlot.tsx

```diff
@@ -85,12 +86,15 @@ const XYPlot: FC<Props> = ({
   },
   theme,
 }) => {
-  const tooltipOpacity = useMemo(() => {
-    if (isFlagEnabled('legendOrientation')) {
-      return legendOpacity
-    }
-    return LEGEND_OPACITY_DEFAULT
-  }, [legendOpacity])
+  const tooltipOpacity = Math.max(
+    useMemo(() => {
+      if (isFlagEnabled('legendOrientation')) {
+        return legendOpacity
+      }
+      return LEGEND_OPACITY_DEFAULT
+    }, [legendOpacity]),
+    LEGEND_OPACITY_MINIMUM
+  )
```

Comment:
All of these are doing the exact same thing.  I would pull them out into their own hook with a `useCallback` whose dependency array is contains `legendOpacity`.  Then each of these files would read something like: 

`const tooltipOpacity = useLegendOpacity(legendOpacity)`

Reply:
Ok, will do.

Also, unrelated. A better default is a solid looking tooltip rather than a barely visible 0.2